gartan_fetch.py

```
@@ -42,6 +42,23 @@ def fetch_and_cache_grid_html(
+    if not session:
+        log_debug(
+            "error",
+            f"No authenticated session provided for grid HTML fetch: {booking_date}",
+        )
+        # Try to use existing cache if it exists, even if expired, as a fallback
+        cache_file = os.path.join(
+            cache_dir, f"grid_{booking_date.replace('/', '-')}.html"
+        )
+        if os.path.exists(cache_file):
+            try:
+                with open(cache_file, "r", encoding="utf-8") as f:
```

Ruff (0.15.2)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)